### PR TITLE
Deltapad code update

### PR DESCRIPTION
**Pull request by rich239**
## Types of Changes

- [x] Core
- [x] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by fauxpark** [comment]
You have some merge conflicts here, because you're trying to add a keyboard that already exists in the repo.

File: keyboards/deltapad/readme.md [hunk 1]
```diff
@@ -1,28 +1,29 @@
...
-
-* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
-
+## DELTAPAD
+
+![DeltaPad](https://github.com/rich239/JPGS/blob/main/deltapad.jpg)
+
+
+*A short description of the keyboard/project*
+
```
Comment (fauxpark) [inline keyboards/deltapad/readme.md:7]:
```suggestion
```

File: keyboards/deltapad/readme.md [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
-
-See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
-
-## Bootloader
-
-Enter the bootloader in the following way:
-
-* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
-
+## DELTAPAD
```
Comment (fauxpark) [inline keyboards/deltapad/readme.md:1]: Please use Unix line endings for all files.

File: keyboards/deltapad/readme.md [hunk 3]
```diff
@@ -1,28 +1,29 @@
...
+
+![DeltaPad](https://github.com/rich239/JPGS/blob/main/deltapad.jpg)
```
Comment (fauxpark) [inline keyboards/deltapad/readme.md:3]: Please rehost this on imgur.

File: keyboards/deltapad/rules.mk [hunk 4]
```diff
@@ -16,6 +16,7 @@ COMMAND_ENABLE = no         # Commands for debug and configuration
 SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
 # if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
 NKRO_ENABLE = no            # USB Nkey Rollover
-BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
-RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+# RGBLIGHT_DRIVER = WS2812     # WS2811/WS2812 lED Driver Support
```
Comment (fauxpark) [inline keyboards/deltapad/rules.mk:21]:
```suggestion
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
```

File: keyboards/deltapad/rules.mk [hunk 5]
```diff
@@ -16,6 +16,7 @@ COMMAND_ENABLE = no         # Commands for debug and configuration
 SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
 # if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
 NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/deltapad/rules.mk:18]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/deltapad/deltapad.c [hunk 6]
```diff
@@ -1,17 +1,17 @@
...
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */
-
-#include "deltapad.h"
+/* Copyright 2021 Richard Snijder
```
Comment (fauxpark) [inline keyboards/deltapad/deltapad.c:1]:
```suggestion
/* Copyright 2021 Richard Snijder

```
As above.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/deltapad/config.h [hunk 7]
```diff
@@ -58,43 +58,53 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
-//#    define RGBLIGHT_EFFECT_ALTERNATING
+#    define RGBLIGHT_EFFECT_BREATHING
+#    define RGBLIGHT_EFFECT_RAINBOW_MOOD
+#    define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+#    define RGBLIGHT_EFFECT_SNAKE
+#    define RGBLIGHT_EFFECT_KNIGHT
+#    define RGBLIGHT_EFFECT_CHRISTMAS
+#    define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#    define RGBLIGHT_EFFECT_RGB_TEST
+#    define RGBLIGHT_EFFECT_ALTERNATING
```
Comment (drashna) [inline keyboards/deltapad/config.h:89]:
```suggestion
#        define RGBLIGHT_HUE_STEP 8
#        define RGBLIGHT_SAT_STEP 8
#        define RGBLIGHT_VAL_STEP 8
#        define RGBLIGHT_LIMIT_VAL 255 /* The maximum brightness level */
#        define RGBLIGHT_SLEEP  /* If defined, the RGB lighting will be switched off when the host goes to sleep */
#        define RGBLIGHT_EFFECT_BREATHING
#        define RGBLIGHT_EFFECT_RAINBOW_MOOD
#        define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#        define RGBLIGHT_EFFECT_SNAKE
#        define RGBLIGHT_EFFECT_KNIGHT
#        define RGBLIGHT_EFFECT_CHRISTMAS
#        define RGBLIGHT_EFFECT_STATIC_GRADIENT
#        define RGBLIGHT_EFFECT_RGB_TEST
#        define RGBLIGHT_EFFECT_ALTERNATING
```

File: keyboards/deltapad/keymaps/default/config.h [hunk 8]
```diff
@@ -0,0 +1,19 @@
+#pragma once
```
Comment (drashna) [inline keyboards/deltapad/keymaps/default/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/deltapad/keymaps/default/keymap.c [hunk 9]
```diff
@@ -1,91 +1,155 @@
...
 #include QMK_KEYBOARD_H
 
-uint16_t copy_paste_timer;
-uint16_t enter_timer;
-
 //extern rgblight_config_t rgblight_config;
+#if defined(RGBLIGHT_ENABLE) && defined(RGBLIGHT_LAYERS)
+static uint32_t rgb_preview_timer = 0;
+#endif
+extern rgblight_config_t rgblight_config;
```
Comment (drashna) [inline keyboards/deltapad/keymaps/default/keymap.c:24]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/deltapad/readme.md [hunk 10]
```diff
@@ -1,14 +1,16 @@
-# deltapad
+## DELTAPAD
```
Comment (fauxpark) [inline keyboards/deltapad/readme.md:1]:
```suggestion
# DELTAPAD
```

File: keyboards/deltapad/readme.md [hunk 11]
```diff
@@ -1,14 +1,16 @@
...
 
+![DeltaPad](https://i.imgur.com/uWlOiU4.jpeg)
 
 
 *A short description of the keyboard/project*
```
Comment (fauxpark) [inline keyboards/deltapad/readme.md:6]:
```suggestion
```

File: keyboards/deltapad/readme.md [hunk 12]
```diff
@@ -1,14 +1,16 @@
...
 
-A 4x4 keypad 
+A 4x4 keypad
 
 * Keyboard Maintainer: [Richard Snijder](https://github.com/rich239)
 * Hardware Supported: Atmel 32u4 based keypads
 
+![DeltaPad](https://i.imgur.com/YClI9dV.jpg)
```
Comment (fauxpark) [inline keyboards/deltapad/readme.md:13]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by rich239** [comment]
I Think  there might be a  fluke of some sort going on Github, In the suggested changes I'm seeing the same functions commented and uncommented, but also more then once. Don't know, but I have an idea this should not occur.

**Comment by fauxpark** [comment]
What do you mean? Could you show a screenshot or something?

**Comment by rich239** [comment]
It's a bit blurry, but it think you can see the isssue

Github

**Comment by fauxpark** [comment]
That is called a diff. It's showing that if the suggestion is committed, the lines in red will be removed, and the lines in green added instead.

File: keyboards/deltapad/rules.mk [hunk 13]
```diff
@@ -15,7 +15,7 @@ COMMAND_ENABLE = no         # Commands for debug and configuration
 # Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
 SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
 # if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (fauxpark) [inline keyboards/deltapad/rules.mk:17]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
This has a merge conflict that needs resolution.

**Pull request closed without merging**